portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -19,14 +19,131 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.FetchOptions; 
+import com.google.appengine.api.datastore.FetchOptions.Builder; 
+import com.google.gson.Gson;
+import com.google.sps.data.Quote; 
+import java.util.List; 
+import java.util.ArrayList; 
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that returns some example content. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private static final String QUOTE = "Quote";
+  private static final String TEXT = "text";
+  private static final String TIMESTAMP = "timestamp"; 
+  private static final int DEFAULT_NUM_QUOTES = 5;
+
+  /**
+    * Retrives all quotes that are stored in Datastore
+    * and puts them into a list of quotes. This list is then written
+    * as the response from the server. 
+    */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query = new Query("Quote").addSort("timestamp", SortDirection.DESCENDING);
+    PreparedQuery quotes = datastore.prepare(query);
+
+    // Get the parameter for number to display, with 5 as the default value.
+    int numQuoteToDisplay = getNumQuoteDisplayed(request, DEFAULT_NUM_QUOTES);
+   
+    // Add each quote to the quote list whose content will be written as 
+    // the response from the servlet. 
+    List<Quote> quoteList = new ArrayList<>();
+    quotes.asList(FetchOptions.Builder.withLimit(numQuoteToDisplay))
+          .forEach((quoteEntity) -> {
+            quoteList.add(extractQuoteFromEntity(quoteEntity));
+          });
+
+    String quotesJson = convertToJson(quoteList);
+    response.setContentType("application/json");
+    response.getWriter().println(quotesJson);
+  }
+
+  /**
+   * Receives the quote that the user has inputed in the survey bar.
+   * This new quote is changed into an Entity that is put into Datastore.
+   * Each quote entity has a "text" and a "timstamp" field. 
+   */ 
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {    
+    String quote = request.getParameter("quote");
+
+    // Only update the new quote if it is not an empty string. 
+    if (quote.length() > 0) {
+      long timestamp = System.currentTimeMillis();
```

Comment:
Let's indicate the time unit in this variable as well.

ex:
timestampMillis

Reply:
Done.